packages/ai/src/ONNXSegmentationController.ts

```diff
@@ -601,6 +644,74 @@ export default class ONNXSegmentationController {
     if (desiredImage.imageId === currentImage?.imageId) {
       return;
     }
+
+    if (this._enabled && this._autoSegmentMode) {
+      if (
+        'isInAcquisitionPlane' in viewport &&
+        !viewport.isInAcquisitionPlane()
+      ) {
+        console.warn(
+          'Non acquisition plane viewports and auto segment mode is not yet supported'
+        );
+        return;
+      }
+
+      const segmentation =
+        cornerstoneTools.segmentation.activeSegmentation.getActiveSegmentation(
+          viewport.id
+        );
+
+      const segmentIndex =
+        cornerstoneTools.segmentation.segmentIndex.getActiveSegmentIndex(
+          segmentation.segmentationId
+        );
+      // Get the labelmap for the previous image
+      const imageIds = viewport.getImageIds();
+      const imageIdIndex = viewport.getCurrentImageIdIndex();
+      const previousImageId = imageIds[imageIdIndex - 1];
+      const nextImageId = imageIds[imageIdIndex + 1];
+
+      if (segmentation) {
+        const previousLabelmapImage =
+          cache.getImageByReferencedImageId(previousImageId);
+
+        const previousLabelmapVoxelManager =
+          previousLabelmapImage?.voxelManager;
+
+        const nextLabelmapImage =
+          cache.getImageByReferencedImageId(nextImageId);
+        const nextLabelmapVoxelManager = nextLabelmapImage?.voxelManager;
+
+        if (previousLabelmapVoxelManager || nextLabelmapVoxelManager) {
+          const pointLists = [];
+          previousLabelmapVoxelManager?.forEach(({ value, pointIJK }) => {
+            if (value === segmentIndex) {
+              const worldCoords = utilities.imageToWorldCoords(
+                previousLabelmapImage.imageId,
+                [pointIJK[0], pointIJK[1]]
+              );
+              pointLists.push(worldCoords);
+            }
+          });
+
+          nextLabelmapVoxelManager?.forEach(({ value, pointIJK }) => {
+            if (value === segmentIndex) {
+              const worldCoords = utilities.imageToWorldCoords(
+                nextLabelmapImage.imageId,
+                [pointIJK[0], pointIJK[1]]
+              );
+              pointLists.push(worldCoords);
+            }
+          });
+
+          // Pick random points from the pointLists
+          this.randomPoints = pointLists
```

Comment:
This seems a little too random - it can end up picking points really close to the edge.  It also feels like it should calculate some number of possible points based on how many are available.  Finally, adding negative points outside a reasonable distance would be useful as it seems pretty aggressive about selecting points.